File: Army(c++)/Warriors/WereWolf.cpp
#include "WereWolf.hpp"

WereWolf::WereWolf(const std::string& name, int hp, int dmg) 
    : Unit(name, hp, dmg, new WereWolfState(this))
    , warrior(WarriorID::WEREWOLF) {}

WereWolf::~WereWolf() {}

WarriorID WereWolf::getWarriorID() const {
    return this->warrior;
}

void WereWolf::setWarriorID(WarriorID newID) {
    this->warrior = newID;
}

File: Army(java)/Warriors/WereWolf.java
package Warriors;

import EnumID.*;
import Basic.*;
import State.*;

public class WereWolf extends Unit {
    private WarriorID warrior;
    
    public WereWolf(String name, int hp, int dmg) {
        super(name, hp, dmg);
        this.warrior = WarriorID.WEREWOLF;
        this.setState(new WereWolfState(this));
    }

    public WarriorID getWarriorID() {
        return warrior;
    }

    public void setWarriorID(WarriorID newID) {
        warrior = newID;
    }
}